lib/service/service.go

```diff
@@ -389,10 +391,9 @@ type TeleportProcess struct {
 	// closed all the listeners)
 	listenersClosed bool
 
-	// forkedPIDs is a collection of a teleport processes forked
-	// during restart used to collect their status in case if the
-	// child process crashed.
-	forkedPIDs []int
+	// forkedTeleportCount is the count of forked Teleport child processes
+	// currently active, as spawned by SIGHUP or SIGUSR2.
+	forkedTeleportCount atomic.Int32
```

Comment:
Is `atomic.Int32` guaranteed to by 32-bit aligned?

Reply:
It's guaranteed to be aligned to whatever the architecture needs; it was never a problem for `int32`, but `atomic.Int64` can be overaligned compared to `int64` in certain architectures (like 32-bit arm) where atomic 64 bit operations require an alignment of 8 but normal operations (and thus the type) only require 4.

Reply:
`atomic.Int64` includes an `atomic.align64` which is special cased by the compiler 😩 

```go
// align64 may be added to structs that must be 64-bit aligned.
// This struct is recognized by a special case in the compiler
// and will not work if copied to any other package.
type align64 struct{}
```